### PR TITLE
feat: add apis to load configuration from json

### PR DESCRIPTION
**Pull request by laiiihz**
```dart
  // some language textmate grammar json string
  const fooJson = "......";
  Highlighter.addLanguage("foo", fooJson);

  // some theme configuration json string
  const barTheme = "......";
  final theme = HighlighterTheme.fromConfiguration(
      barTheme, const TextStyle(color: Colors.black));

  final fooHighligher = Highlighter(language: 'foo', theme: theme);

  const fooCode = "......";
  final span = fooHighligher.highlight(fooCode);
```

### CODE REVIEW

File: lib/src/highlighter.dart [hunk 1]
```diff
@@ -176,6 +185,14 @@ class HighlighterTheme {
 
   HighlighterTheme._({required TextStyle wrapper}) : _wrapper = wrapper;
 
+  /// load a [HighlighterTheme] from a [json] string.
```
Comment (vlidholt) [inline lib/src/highlighter.dart:188]: Just a tiny fix to the formatting of the comment.

```dart
/// Load a [HighlighterTheme] from a JSON string.
```

Missing captial L and json should be in brackets as it is not referring to a Dart entity.

File: lib/src/highlighter.dart [hunk 2]
```diff
@@ -176,6 +185,14 @@ class HighlighterTheme {
...
+  factory HighlighterTheme.fromConfiguration(
+      String json, TextStyle defaultStyle) {
```
Comment (vlidholt) [inline lib/src/highlighter.dart:190]: Please add a trailing comma for readability.

File: lib/src/highlighter.dart [hunk 3]
```diff
@@ -54,6 +54,15 @@ class Highlighter {
     }
   }
 
+  /// Adds a custom language to the list of languages.
+  ///
+  /// Format in json string and follow the textmate bundle format.
+  ///
+  /// This must be called before creating any [Highlighter]s.
```
Comment (vlidholt) [inline lib/src/highlighter.dart:61]: Better documentation:
```dart
/// Adds a custom language to the list of languages. Associates a language [name] with a TextMate formatted [json] definition.
```

**Review by vlidholt — CHANGES_REQUESTED** [review]
Thanks a lot for this contribution, just a few minor fixes and it's good to go! :D

**Comment by laiiihz** [comment]
thanks for review! i made some commits to this pr 😆

File: lib/src/highlighter.dart (hunk 1 above)
Comment (laiiihz) [reply lib/src/highlighter.dart]: sorry for missing this changes 🙏

**Comment by Skyost** [comment]
Is there any news on this one ?

**Comment by laiiihz** [comment]
@vlidholt sorry to bother you, but could you please re-review this PR❤️

**Comment by vlidholt** [comment]
Sorry, I missed this. Will have a look soon! Thanks again!

**Review by vlidholt — APPROVED** [review]
LGTM :)

**Pull request merged**